### PR TITLE
MM-18784 Removed deprecated methods

### PR DESCRIPTION
**Pull request by ChrisDobby**
#### Summary
Replaces the use of componentWillReceiveProps in the following files:

components/permalink_view/permalink_view.jsx
components/channel_view/channel_view.jsx
components/rhs_thread/rhs_thread.jsx
components/post_view/new_messages_below/new_messages_below.jsx

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12341

### CODE REVIEW

File: components/channel_view/channel_view.jsx [hunk 1]
```diff
@@ -86,6 +80,10 @@ export default class ChannelView extends React.PureComponent {
     }
 
     componentDidUpdate(prevProps) {
+        if (prevProps.match.url !== this.props.match.url) {
+            this.createDeferredPostView();
```
[the remark this answers is not in the document]
Comment (jwilander) [reply components/channel_view/channel_view.jsx]: Yes, needs to be called before render. I think moving it into state and using `getDerivedStateFromProps`  makes sense 👍

**Comment by ChrisDobby** [comment]
Thanks @hmhealey I should get chance to look at this shortly and if I have any questions I'll let you know 👍

File: components/post_view/new_messages_below/new_messages_below.jsx [hunk 2]
```diff
@@ -36,6 +36,12 @@ export default class NewMessagesBelow extends React.PureComponent {
         }
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps !== this.props) {
+            this.updateFromProps();
```
[the remark this answers is not in the document]
Comment (ChrisDobby) [reply components/post_view/new_messages_below/new_messages_below.jsx]: I may be wrong here but I don't see the use of the rendered state?  I've removed both the visible and rendered state and calculated the visibility in render and everything seems OK when using the application.

**Comment by hmhealey** [comment]
Thanks for changing that and fixing that test. Can you fix the merge conflicts on the PR so we can take a final look at this and get it merged?

**Comment by ChrisDobby** [comment]
Conflicts should be fixed now

**Pull request merged**